packages/request-handler/src/docker.ts

```diff
@@ -0,0 +1,114 @@
+import dayjs from "dayjs";
+import Docker from "dockerode";
+import { z } from "zod";
+
+import { db, like, or } from "@homarr/db";
+import { icons } from "@homarr/db/schema";
+
+import type { ContainerState } from "../../docker/src";
+import { createCachedWidgetRequestHandler } from "./lib/cached-widget-request-handler";
+
+interface ContainerInfo {
+  Id: string;
+  Names: string[];
+  State: string;
+  Status: string;
+  Image: string;
+}
+
+export const dockerContainersRequestHandler = createCachedWidgetRequestHandler({
+  queryKey: "dockerContainersResult",
+  widgetKind: "dockerContainers",
+  async requestAsync() {
+    const containers = await getContainersWithStats();
+
+    return responseSchema.parse({ containers });
+  },
+  cacheDuration: dayjs.duration(5, "seconds"),
+});
+
+const responseSchema = z.object({
+  containers: z.array(
+    z.object({
+      id: z.string(),
+      name: z.string(),
+      state: z.string(),
+      status: z.string(),
+      iconUrl: z.string().nullable(),
+      cpuUsage: z.number(),
+      memoryUsage: z.number(),
+    }),
+  ),
+});
+
+export type DockerContainersStatus = z.infer<typeof responseSchema>;
+const docker = new Docker();
```

Comment:
Use docker singleton or add some way the user can choose which of the configured docker instances should be used